### PR TITLE
Bump OBO and move ScrapeConfig to structured

### PR DESCRIPTION
**Pull request by vyzigold**
We still can go a few versions further once we bump controller-runtime again, but this gets us almost caught up with OBO. This updates to OBO v0.0.28. The most recent is OBO v0.1.0, which seems to come after v0.0.30, but these need controller-runtime 17.

### CODE REVIEW

**Comment by vyzigold** [comment]
/retest
failed to pull images:
```
 INFO[2024-03-21T08:31:34Z] Build openstack-operator-ci-build-deploy-kuttl-amd64 failed, printing logs: 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100     2  100     2    0     0     31      0 --:--:-- --:--:-- --:--:--    31
Caching blobs under "/var/cache/blobs".
Trying to pull image-registry.openshift-image-registry.svc:5000/ci-op-l2zcnmdf/pipeline@sha256:727609367291999cc870353ac2291c87c970498314978db2b563d0f72afa2f97...
error: error creating buildah builder: initializing source docker://image-registry.openshift-image-registry.svc:5000/ci-op-l2zcnmdf/pipeline@sha256:727609367291999cc870353ac2291c87c970498314978db2b563d0f72afa2f97: reading manifest sha256:727609367291999cc870353ac2291c87c970498314978db2b563d0f72afa2f97 in image-registry.openshift-image-registry.svc:5000/ci-op-l2zcnmdf/pipeline: manifest unknown 
```

**Comment by vyzigold** [comment]
recheck
```
TASK [cert_manager : Verify cert_manager api _raw_params={{ lookup('env', 'HOME') }}/bin/cmctl check api --wait=2m] ***
Thursday 21 March 2024  10:15:03 -0400 (0:00:02.116)       0:31:48.400 ******** 
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["/home/zuul/bin/cmctl", "check", "api", "--wait=2m"], "delta": "0:02:00.069886", "end": "2024-03-21 10:17:03.288956", "msg": "non-zero return code", "rc": 124, "start": "2024-03-21 10:15:03.219070", "stderr": "", "stderr_lines": [], "stdout": "error: the cert-manager CRDs are not yet installed on the Kubernetes API server", "stdout_lines": ["error: the cert-manager CRDs are not yet installed on the Kubernetes API server"]}

```

**Review by csibbitt — APPROVED** [review]
LGTM - I have questions about why we are still tracking OBO and not COO, but I'll try to get that answered out of band.

File: api/bases/telemetry.openstack.org_metricstorages.yaml [hunk 1]
```diff
@@ -902,11 +951,15 @@ spec:
...
+                        pattern: ^(0|(([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?)$
+                        type: string
                     type: object
                   resourceSelector:
-                    description: Label selector for Monitoring Stack Resources. Set
-                      to the empty LabelSelector ({}) to monitoring everything. Set
-                      to null to disable service discovery.
+                    description: 'Label selector for Monitoring Stack Resources. To
+                      monitor everything, set to empty map selector. E.g. resourceSelector:
+                      {}. To disable service discovery, set to null. E.g. resourceSelector:.'
```
[the remark this answers is not in the document]
Comment (vyzigold) [reply api/bases/telemetry.openstack.org_metricstorages.yaml:962]: yes

**Comment by vyzigold** [comment]
recheck
cert manager again

**Comment by vyzigold** [comment]
recheck
failure in content provider

**Comment by vyzigold** [comment]
recheck
content provider job should be fixed

**Comment by vyzigold** [comment]
recheck
timing issue between ceilometer and keystone
```
2024-03-25 15:54:25.816 14 ERROR ceilometer.polling.manager [-] Skipping images, keystone issue: internalURL endpoint for image service not found: keystoneauth1.exceptions.catalog.EndpointNotFound: internalURL endpoint for image service not found
2024-03-25 15:54:25.817 14 DEBUG ceilometer.polling.manager [-] Skip pollster image.size, no  resources found this cycle poll_and_notify /usr/lib/python3.9/site-packages/ceilometer/polling/manager.py:193
```

**Comment by vyzigold** [comment]
recheck

**Comment by vyzigold** [comment]
recheck
Unrelated error
```
tempest.exceptions.BuildErrorException: Server db943723-e98f-46bd-ad38-442eafff8ef4 failed to build and is in ERROR status
Details:  Request ID of server operation performed before checking the server status req-a6ab0263-c255-4fca-b27f-50b1c8e4553f.
```

**Pull request merged**